Hyper/MarchingCubes/ChunkWorker.cs

```diff
@@ -0,0 +1,114 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Reflection.Metadata;
+using System.Threading;
+using Hyper.Collisions;
+using Hyper.MarchingCubes;
+using Hyper.Meshes;
+using OpenTK.Mathematics;
+using Hyper.Collisions.Bepu;
+
+namespace Hyper
+{
+    internal class ChunkWorker
+    {
+        private ConcurrentDictionary<Vector3i, Guid> _neededChunkPositions;
+        private ConcurrentDictionary<Vector3i, Guid> _existingChunkPositions;
+        private ConcurrentDictionary<Guid, Chunk> _existingChunks;
+        private CancellationTokenSource _cancellationTokenSource;
+        private ScalarFieldGenerator _scalarFieldGenerator;
+        private ChunkFactory _chunkFactory;
+        private SimulationManager<NarrowPhaseCallbacks, PoseIntegratorCallbacks> _simulationManager;
+
+        public ChunkWorker(ConcurrentDictionary<Guid, Chunk> existingChunks, ScalarFieldGenerator scalarFieldGenerator, 
+            SimulationManager<NarrowPhaseCallbacks, PoseIntegratorCallbacks> simulationManager)
+        {
+            _cancellationTokenSource = new CancellationTokenSource();
+            _neededChunkPositions = new ConcurrentDictionary<Vector3i, Guid>();
+            _neededChunkPositions.TryAdd(new Vector3i(0, 0, 0), Guid.NewGuid()); // initial chunk...
+            _existingChunkPositions = new ConcurrentDictionary<Vector3i, Guid>();
+            _existingChunks = existingChunks;
+            _scalarFieldGenerator = scalarFieldGenerator;
+            _chunkFactory = new ChunkFactory(_scalarFieldGenerator);
+            _simulationManager = simulationManager;
+        }
+        public void StartProcessing()
+        {
+            var processingThread = new Thread(ProcessItems);
```

Comment:
You can do this:
var processingThread = new Thread(ProcessItems)
            {
                IsBackground = true
            };